docs/deep-dives/esm.md

```diff
@@ -145,9 +145,13 @@ Your local `tsconfig.json` files will require a `baseUrl` and `rootDir` to help
 This allows us to import modules like this:
 
 ```ts
-import { module } from '#src/imported-module';
+import { module } from '#src/imported-module.js';
 ```
 
+[Custom conditions] enable tooling to use TypeScript source files during development while automatically switching to the correct compiled outputs for deployment or publishing. This is done without additional configuration to modify package.json imports and exports. Monorepo users may already be familiar with this concept through pnpm's `publishConfig` feature. For more details, see [Live types in a TypeScript monorepo].
```

Comment:
Lazy question: does tsc rewrite `customConditions` during transpile, or are we now relying on [`node --conditions`](https://nodejs.org/api/cli.html#-c-condition---conditionscondition)?

Reply:
ah right!